kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -28,6 +29,23 @@ const fetchFullContent = ({ dispatch }, id) => {
 const nodeAssignment = (nodes, dispatch) => {
   const topics = nodes.filter((node) => node.kind === 'topic');
   const contents = nodes.filter((node) => node.kind !== 'topic');
+
+  // clean up API response
+  contents.forEach(content => {
+    if (!content.progress) {
+      logging.warn('"progress" was not included in API response');
+      content.progress = 'unstarted'; // eslint-disable-line no-param-reassign
```

Comment:
I would rather handle this with server side annotation and graceful defaults in the rendering code.